src/main/java/com/google/devtools/build/docgen/templates/be/make-variables.vm

```diff
@@ -125,112 +141,347 @@ in your genrule's cmd attribute.
+  </li>
+  <li><code>COMPILATION_MODE</code>:
+    "fastbuild", "dbg", or "opt". (<a
+  href="https://docs.bazel.build/versions/master/user-manual.html#flag--compilation_mode">more
+  details</a>)
   </li>
-  <li><code>COMPILATION_MODE</code>: "fastbuild", "dbg", or "opt".</li>
-
 </ul>
 
-<p><strong>Path Variables</strong></p>
+<p><strong>Path variables</strong></p>
 
 <ul><!--  keep alphabetically sorted  -->
-  <li><code>BINDIR</code>: The base of the generated binary tree for the target
-    architecture.  (Note that a different tree may be used for
-    programs that run during the build on the host architecture,
-    to support cross-compiling.  If you want to run a tool from
-    within a genrule, the recommended way of specifying the path to
-    the tool is to use <code>$(location <i>toolname</i>)</code>,
-    where <i>toolname</i> must be listed in the <code>tools</code>
-    attribute for the genrule.</li>
-  <li><code>GENDIR</code>: The base of the generated code
-    tree for the target architecture.</li>
-  <li class="harmful"><code>JAVABASE</code>: for Java, most of the tools in the
-    JDK should not be used as-is. The built-in Java rules use much more
-    sophisticated approaches to Java compilation and packaging than the upstream
-    tools can express, such as interface Jars, header interface Jars, and highly
-    optimized Jar packaging and merging implementations.
-
-    <p>The base directory containing the Java utilities. May be a relative path.
-    It will have a "bin" subdirectory.</p>
-
-    <p>This variable is only available if the Java runtime, that is,
-
-<code>@bazel_tools//tools/jdk:current_java_runtime</code>
-    or its host version (<code>current_host_java_runtime</code>) is in the
-    <code>toolchains</code> attribute.</p>
+  <li>
+    <p>
+      <code>BINDIR</code>: The base of the generated binary tree for the target
+       architecture.
+    </p>
+
+    <p>
+      Note that a different tree may be used for programs that run during the
+      build on the host architecture, to support cross-compiling.
+    </p>
+
+    <p>
+      If you want to run a tool from within a <code>genrule</code>, the
+      recommended way to get its path is <code>($<a
+      href="#predefined_label_variables">execpath</a> toolname)</code>,
+      where <i>toolname</i> must be listed in the <code>genrule</code>'s
+      <code><a
+      href="$expander.expandRef("genrule.tools")">tools</a></code> attribute.
+    </p>
+  </li>
+      
+  <li><code>GENDIR</code>:
+    The base of the generated code tree for the target architecture.
   </li>
 </ul>
 
-<p><strong>Architecture Variables</strong></p>
+<p><strong>Machine architecture variables</strong></p>
 
 <ul><!--  keep alphabetically sorted  -->
   <li><code>ABI</code>:
     The C++ ABI version. </li>
 
-  <li> <code>TARGET_CPU</code>: The target architecture's cpu,
-    e.g. "piii" or "k8". </li>
+  <li> <code>TARGET_CPU</code>: The target architecture's CPU, e.g. "k8". </li>
```

Comment:
TARGET_CPU is no longer a C++ make variable: it really is globally provided.

Reply:
What do you mean? As currently expressed it's listed as a global "Machine architecture variable", not C++-specific.

Reply:
Oh, sorry, I misread the location because reading raw HTML melts my brain.